diff --git a/IOSupport/glsl.cpp b/IOSupport/glsl.cpp
index 61cbc63..aec8b1f 100644
--- a/IOSupport/glsl.cpp
+++ b/IOSupport/glsl.cpp
@@ -3,18 +3,6 @@
 
 #if 1
 #include <glad.h>
-#define HAVE_GLU 1
-#if defined(_WIN32) || defined(__WIN32__) || defined(WIN32)
-// https://github.com/NatronGitHub/openfx-io/issues/4#issuecomment-997284560
-#undef HAVE_GLU
-#endif
-#if HAVE_GLU
-#ifdef __APPLE__
-#include <OpenGL/glu.h>
-#else
-#include <GL/glu.h>
-#endif
-#endif
 #else
 #ifdef __APPLE__
 
@@ -42,6 +30,8 @@
 
 #include "glsl.h"
 
+#include "ofxsOGLDebug.h" // glErrorString()
+
 namespace OCIO_NAMESPACE {
 
 namespace {
@@ -49,11 +39,7 @@ namespace {
     {
         const GLenum glErr = glGetError();
         if (glErr != GL_NO_ERROR) {
-#if HAVE_GLU
-            error = (const char*)gluErrorString(glErr);
-#else
-            error = "OpenGL Error";
-#endif
+            error = (const char*)glErrorString(glErr);
             return true;
         }
         return false;
