diff -aru build_scripts/utils.py build_scripts/utils.py
--- build_scripts/utils.py	2024-07-24 13:29:26.697629300 -0700
+++ build_scripts/utils.py	2024-07-29 12:24:51.747387000 -0700
@@ -55,14 +55,16 @@
 except ImportError:
     import urllib
 
-import distutils.log as log
-from distutils.errors import DistutilsSetupError
+#import distutils.log as log
+import logging
+#from distutils.errors import DistutilsSetupError
 
 try:
     WindowsError
 except NameError:
     WindowsError = None
 
+log = logging.getLogger("blah")
 
 def is_64bit():
     return sys.maxsize > 2147483647
@@ -119,7 +121,7 @@
                     continue
                 setenv_paths.append(setenv_path)
     if len(setenv_paths) == 0:
-        raise DistutilsSetupError("Failed to find the Windows SDK with MSVC compiler "
+        raise Error("Failed to find the Windows SDK with MSVC compiler "
                                   "version {}".format(MSVC_VERSION))
     for setenv_path in setenv_paths:
         log.info("Found {}".format(setenv_path))
diff -aru sources/cmake_helpers/helpers.cmake sources/cmake_helpers/helpers.cmake
--- sources/cmake_helpers/helpers.cmake	2024-07-24 13:29:27.981724200 -0700
+++ sources/cmake_helpers/helpers.cmake	2024-07-29 12:24:51.747387000 -0700
@@ -47,7 +47,7 @@
     Charts
     Svg
     DataVisualization)
-find_package(Qt${QT_MAJOR_VERSION}UiTools)
+find_package(Qt${QT_MAJOR_VERSION} COMPONENTS UiTools)
 if(Qt${QT_MAJOR_VERSION}UiTools_FOUND)
     list(APPEND ALL_OPTIONAL_MODULES UiTools)
 else()
@@ -118,10 +118,10 @@
 endmacro()
 
 macro(COLLECT_MODULE_IF_FOUND shortname)
-    set(name "Qt${QT_MAJOR_VERSION}${shortname}")
+    set(name "${shortname}")
     set(_qt_module_name "${name}")
     if ("${shortname}" STREQUAL "OpenGLFunctions")
-        set(_qt_module_name "Qt${QT_MAJOR_VERSION}Gui")
+        set(_qt_module_name "Gui")
     endif()
     # Determine essential/optional/missing
     set(module_state "missing")
@@ -141,7 +141,7 @@
         set(quiet_argument "QUIET")
     endif()
 
-    find_package(${_qt_module_name} ${quiet_argument})
+    find_package(Qt${QT_MAJOR_VERSION} COMPONENTS ${_qt_module_name} ${quiet_argument})
     # If package is found, _name_found will be equal to 1
     set(_name_found "${_qt_module_name}_FOUND")
     # _name_dir will keep the path to the directory where the CMake rules were found
diff -aru sources/pyside2/CMakeLists.txt sources/pyside2/CMakeLists.txt
--- sources/pyside2/CMakeLists.txt	2024-07-24 13:29:29.952545500 -0700
+++ sources/pyside2/CMakeLists.txt	2024-07-29 12:24:51.747387000 -0700
@@ -17,6 +17,12 @@
 include(shiboken_helpers)
 include(helpers)
 
+if (USE_PYTHON_VERSION)
+    shiboken_find_required_python(${USE_PYTHON_VERSION})
+else()
+    shiboken_find_required_python()
+endif()
+
 # Don't display "up-to-date / install" messages when installing, to reduce visual clutter.
 if (QUIET_BUILD)
     set(CMAKE_INSTALL_MESSAGE NEVER)
@@ -41,12 +47,12 @@
   ${PYSIDE_VERSION_FILE_PATH}
 )
 execute_process(
-  COMMAND ${PYTHON_EXECUTABLE} "${PYSIDE_VERSION_FILE_PATH}"
+  COMMAND ${Python3_EXECUTABLE} "${PYSIDE_VERSION_FILE_PATH}"
   OUTPUT_VARIABLE PYSIDE_VERSION_OUTPUT
   ERROR_VARIABLE PYSIDE_VERSION_OUTPUT_ERROR
   OUTPUT_STRIP_TRAILING_WHITESPACE)
 if (NOT PYSIDE_VERSION_OUTPUT)
-    message(FATAL_ERROR "Could not identify PySide2 version. Error: ${PYSIDE_VERSION_OUTPUT_ERROR}")
+    message(FATAL_ERROR "Could not identify PySide2 version. Error: '${PYSIDE_VERSION_OUTPUT_ERROR}'")
 endif()
 
 list(GET PYSIDE_VERSION_OUTPUT 0 BINDING_API_MAJOR_VERSION)
@@ -79,7 +85,7 @@
     message(STATUS "******************************************************")
 endif()
 
-find_package(Qt${QT_MAJOR_VERSION} 5.12 REQUIRED COMPONENTS Core)
+find_package(Qt${QT_MAJOR_VERSION} 5.15 REQUIRED COMPONENTS Core)
 add_definitions(${Qt${QT_MAJOR_VERSION}Core_DEFINITIONS})
 
 find_file(GL_H "gl.h" PATH_SUFFIXES "GL")
@@ -117,15 +123,10 @@
 endif()
 set(CMAKE_CXX_STANDARD_REQUIRED ON)
 
-# >= Qt5: QT_INCLUDE_DIR does no longer exist. Derive from QtCore
-if(${QT_MAJOR_VERSION} GREATER_EQUAL 6)
-    get_target_property(QT_INCLUDE_DIR Qt6::Core INTERFACE_INCLUDE_DIRECTORIES)
-    get_filename_component(QT_INCLUDE_DIR "${QT_INCLUDE_DIR}" DIRECTORY)
-else()
-    # On Windows, macOS, and Linux it can be computed from Qt5Core_INCLUDE_DIRS, which contains
-    # a list of include directories. We take the first one.
-    list(GET Qt5Core_INCLUDE_DIRS 0 QT_INCLUDE_DIR)
-endif()
+#get_target_property(QT_INCLUDE_DIR Qt${QT_MAJOR_VERSION}::Core INTERFACE_INCLUDE_DIRECTORIES)
+#get_filename_component(QT_INCLUDE_DIR "${QT_INCLUDE_DIR}" DIRECTORY)
+set(QT_INCLUDE_DIR "${Qt5_INCLUDE_DIR}")
+
 message(STATUS "*** Qt ${QT_MAJOR_VERSION}, QT_INCLUDE_DIR= ${QT_INCLUDE_DIR}")
 
 # On macOS, check if Qt is a framework build. This affects how include paths should be handled.
@@ -186,7 +187,7 @@
 endforeach()
 
 
-string(REGEX MATCHALL "[0-9]+" qt_version_helper "${Qt${QT_MAJOR_VERSION}Core_VERSION}")
+string(REGEX MATCHALL "[0-9]+" qt_version_helper "${Qt5_VERSION_STRING}")
 
 list(GET qt_version_helper 0 QT_VERSION_MAJOR)
 list(GET qt_version_helper 1 QT_VERSION_MINOR)
@@ -220,14 +221,7 @@
 endif()
 
 if (NOT PYTHON_SITE_PACKAGES)
-    execute_process(
-        COMMAND ${SHIBOKEN_PYTHON_INTERPRETER} -c "if True:
-            from distutils import sysconfig
-            from os.path import sep
-            print(sysconfig.get_python_lib(1, 0, prefix='${CMAKE_INSTALL_PREFIX}').replace(sep, '/'))
-            "
-        OUTPUT_VARIABLE PYTHON_SITE_PACKAGES
-        OUTPUT_STRIP_TRAILING_WHITESPACE)
+    set_python_site_packages()
     if (NOT PYTHON_SITE_PACKAGES)
         message(FATAL_ERROR "Could not detect Python module installation directory.")
     elseif (APPLE)
diff -aru sources/pyside2/cmake/Macros/FindQt5Extra.cmake sources/pyside2/cmake/Macros/FindQt5Extra.cmake
--- sources/pyside2/cmake/Macros/FindQt5Extra.cmake	2024-07-24 13:29:28.225585400 -0700
+++ sources/pyside2/cmake/Macros/FindQt5Extra.cmake	2024-07-29 12:24:51.747387000 -0700
@@ -4,7 +4,7 @@
 # CT: maybe we can remove this.
 # For now, I just use the mapping to Qt5
 
-find_package(Qt${QT_MAJOR_VERSION}Multimedia)
+find_package(Qt${QT_MAJOR_VERSION} COMPONENTS Multimedia)
 
 if (NOT Qt${QT_MAJOR_VERSION}Multimedia_FOUND)
     find_path(QT_QTMULTIMEDIA_INCLUDE_DIR QtMultimedia
diff -aru sources/pyside2/libpyside/CMakeLists.txt sources/pyside2/libpyside/CMakeLists.txt
--- sources/pyside2/libpyside/CMakeLists.txt	2024-07-24 13:29:29.174034900 -0700
+++ sources/pyside2/libpyside/CMakeLists.txt	2024-07-29 12:24:51.763010400 -0700
@@ -82,8 +82,9 @@
 
 target_include_directories(pyside2 PRIVATE
     ${QML_INCLUDES}
-    ${Qt${QT_MAJOR_VERSION}Core_PRIVATE_INCLUDE_DIRS}
-    ${Qt${QT_MAJOR_VERSION}Core_INCLUDE_DIRS}
+    ${qt_PACKAGE_FOLDER_RELEASE}/include/QtCore/${Qt5_VERSION_STRING}
+    ${qt_PACKAGE_FOLDER_RELEASE}/include/QtCore/${Qt5_VERSION_STRING}/QtCore
+    ${Qt${QT_MAJOR_VERSION}_INCLUDE_DIR}
 )
 
 target_include_directories(pyside2 PUBLIC
@@ -94,7 +95,7 @@
 target_link_libraries(pyside2
                       PRIVATE Shiboken2::libshiboken
                       PRIVATE ${QML_LIBRARIES}
-                      PRIVATE ${Qt${QT_MAJOR_VERSION}Core_LIBRARIES})
+                      PRIVATE qt::qt)
 
 set_target_properties(pyside2 PROPERTIES
                       VERSION ${BINDING_API_VERSION}
diff -aru sources/pyside2-tools/CMakeLists.txt sources/pyside2-tools/CMakeLists.txt
--- sources/pyside2-tools/CMakeLists.txt	2024-07-24 13:29:29.957542700 -0700
+++ sources/pyside2-tools/CMakeLists.txt	2024-07-29 13:24:56.299193600 -0700
@@ -9,6 +9,7 @@
 find_package(Shiboken2 2.0.0 REQUIRED)
 find_package(Qt5 REQUIRED Core)
 find_package(PySide2 2.0.0 REQUIRED)
+find_package(Python3 COMPONENTS Interpreter Development REQUIRED)
 
 set(pyside2_tools_MAJOR_VERSION "2")
 set(pyside2_tools_MINOR_VERSION "0")
@@ -27,10 +28,11 @@
 # UIC stuff
 if (NOT PYTHON_SITE_PACKAGES)
     execute_process(
-        COMMAND ${SHIBOKEN_PYTHON_INTERPRETER} -c "if True:
-            from distutils import sysconfig
+        COMMAND ${Python3_EXECUTABLE} -c "if True:
+            import sysconfig
             from os.path import sep
-            print(sysconfig.get_python_lib(1, 0, prefix='${CMAKE_INSTALL_PREFIX}').replace(sep, '/'))
+            tmp = sysconfig.get_path('platlib').replace(sysconfig.get_path('platstdlib'), '')
+            print(('${CMAKE_INSTALL_PREFIX}'+ tmp).replace(sep, '/') )
             "
         OUTPUT_VARIABLE PYTHON_SITE_PACKAGES
     OUTPUT_STRIP_TRAILING_WHITESPACE)
@@ -46,7 +48,8 @@
     set(EXE_EXT "")
 endif()
 
-set(TOOLS_PATH "${_qt5Core_install_prefix}/bin")
+set(TOOLS_PATH "${qt_PACKAGE_FOLDER_RELEASE}/bin")
+message("BLAP ${TOOLS_PATH}")
 set(UIC_PATH "${TOOLS_PATH}/uic${EXE_EXT}")
 set(RCC_PATH "${TOOLS_PATH}/rcc${EXE_EXT}")
 if (APPLE)
diff -aru sources/pyside2-tools/pylupdate/CMakeLists.txt sources/pyside2-tools/pylupdate/CMakeLists.txt
--- sources/pyside2-tools/pylupdate/CMakeLists.txt	2024-07-24 13:29:29.966558500 -0700
+++ sources/pyside2-tools/pylupdate/CMakeLists.txt	2024-07-29 12:24:51.763010400 -0700
@@ -11,10 +11,7 @@
 translator.cpp
 )
 
-find_package(Qt5Core)
-find_package(Qt5Gui)
-find_package(Qt5Xml)
-find_package(Qt5Widgets)
+find_package(Qt${QT_MAJOR_VERSION} COMPONENTS Core Gui Xml Widgets)
 
 set(lupdate_MOC_HEADERS translator.h)
 qt5_wrap_cpp(lupdate_MOC_OUTFILES ${lupdate_MOC_HEADERS})
@@ -22,17 +19,11 @@
 add_executable(pyside2-lupdate ${lupdate_SRC} ${lupdate_MOC_OUTFILES})
 include_directories(pyside2-lupdate
                     ${CMAKE_CURRENT_SOURCE_DIR}
-                    ${Qt5Xml_INCLUDE_DIRS}
-                    ${Qt5Core_INCLUDE_DIRS}
-                    ${Qt5Gui_INCLUDE_DIRS}
-                    ${Qt5Widgets_INCLUDE_DIRS}
+                    ${Qt${QT_MAJOR_VERSION}_INCLUDE_DIR}
                     )
 
 target_link_libraries(pyside2-lupdate
-                      ${Qt5Core_LIBRARIES}
-                      ${Qt5Xml_LIBRARIES}
-                      ${Qt5Gui_LIBRARIES}
-                      ${Qt5Widgets_LIBRARIES}
+                      qt::qt
                       )
 
 install(TARGETS pyside2-lupdate RUNTIME DESTINATION bin)
diff -aru sources/shiboken2/ApiExtractor/CMakeLists.txt sources/shiboken2/ApiExtractor/CMakeLists.txt
--- sources/shiboken2/ApiExtractor/CMakeLists.txt	2024-07-24 13:29:27.370169800 -0700
+++ sources/shiboken2/ApiExtractor/CMakeLists.txt	2024-07-29 12:24:51.763010400 -0700
@@ -32,8 +32,7 @@
 xmlutils.cpp
 )
 
-find_package(Qt${QT_MAJOR_VERSION}XmlPatterns 5.12)
-find_package(Qt${QT_MAJOR_VERSION}Xml 5.12)
+find_package(Qt${QT_MAJOR_VERSION} COMPONENTS Xml XmlPatterns 5.15)
 find_package(LibXml2 2.6.32)
 find_package(LibXslt 1.1.19)
 
@@ -42,7 +41,7 @@
     set(HAS_LIBXSLT 1)
 endif()
 
-if(NOT Qt${QT_MAJOR_VERSION}XmlPatterns_FOUND AND NOT HAS_LIBXSLT)
+if(NOT Qt${QT_MAJOR_VERSION}XmlPatterns_FOUND AND NOT LibXslt_FOUND)
     set(DISABLE_DOCSTRINGS TRUE)
     message(WARNING
             "Documentation will not be built due to missing dependency (no Qt5XmlPatterns found).")
@@ -59,13 +58,13 @@
 target_link_libraries(apiextractor PUBLIC Qt${QT_MAJOR_VERSION}::Core)
 target_link_libraries(apiextractor PRIVATE ${CLANG_EXTRA_LIBRARIES})
 
-if (HAS_LIBXSLT)
+if (LibXslt_FOUND)
     target_compile_definitions(apiextractor PUBLIC HAVE_LIBXSLT)
     target_sources(apiextractor PRIVATE xmlutils_libxslt.cpp)
     target_include_directories(apiextractor
-                               PRIVATE ${LIBXSLT_INCLUDE_DIR} ${LIBXML2_INCLUDE_DIR})
+                               PRIVATE ${LibXslt_INCLUDE_DIR} ${LibXml2_INCLUDE_DIR})
     target_link_libraries(apiextractor
-                          PRIVATE ${LIBXSLT_LIBRARIES} ${LIBXML2_LIBRARIES})
+                          PRIVATE ${LibXslt_LIBRARIES} ${LibXml2_LIBRARIES})
 endif()
 
 if (Qt${QT_MAJOR_VERSION}XmlPatterns_FOUND)
diff -aru sources/shiboken2/CMakeLists.txt sources/shiboken2/CMakeLists.txt
--- sources/shiboken2/CMakeLists.txt	2024-07-24 13:29:27.970730600 -0700
+++ sources/shiboken2/CMakeLists.txt	2024-07-29 12:24:51.763010400 -0700
@@ -20,7 +20,7 @@
 
 set (QT_MAJOR_VERSION 5)
 message(STATUS "Using Qt ${QT_MAJOR_VERSION}")
-find_package(Qt${QT_MAJOR_VERSION} 5.12 REQUIRED COMPONENTS Core)
+find_package(Qt${QT_MAJOR_VERSION} 5.15 REQUIRED COMPONENTS Core)
 
 if (QUIET_BUILD)
     set_quiet_build()
@@ -43,7 +43,7 @@
   ${SHIBOKEN_VERSION_FILE_PATH}
 )
 execute_process(
-  COMMAND ${PYTHON_EXECUTABLE} "${SHIBOKEN_VERSION_FILE_PATH}"
+  COMMAND ${Python3_EXECUTABLE} "${SHIBOKEN_VERSION_FILE_PATH}"
   OUTPUT_VARIABLE SHIBOKEN_VERSION_OUTPUT
   ERROR_VARIABLE SHIBOKEN_VERSION_OUTPUT_ERROR
   OUTPUT_STRIP_TRAILING_WHITESPACE)
@@ -67,12 +67,12 @@
 
 ## For debugging the PYTHON* variables
 message(STATUS "PYTHONLIBS_FOUND:       " ${PYTHONLIBS_FOUND})
-message(STATUS "PYTHON_LIBRARIES:       " ${PYTHON_LIBRARIES})
-message(STATUS "PYTHON_INCLUDE_DIRS:    " ${PYTHON_INCLUDE_DIRS})
-message(STATUS "PYTHON_DEBUG_LIBRARIES: " ${PYTHON_DEBUG_LIBRARIES})
-message(STATUS "PYTHONINTERP_FOUND:     " ${PYTHONINTERP_FOUND})
-message(STATUS "PYTHON_EXECUTABLE:      " ${PYTHON_EXECUTABLE})
-message(STATUS "PYTHON_VERSION:         " ${PYTHON_VERSION_MAJOR}.${PYTHON_VERSION_MINOR}.${PYTHON_VERSION_PATCH})
+message(STATUS "Python3_LIBRARIES:       " ${Python3_LIBRARIES})
+message(STATUS "Python3_INCLUDE_DIRS:    " ${Python3_INCLUDE_DIRS})
+message(STATUS "Python3_DEBUG_LIBRARIES: " ${Python3_DEBUG_LIBRARIES})
+message(STATUS "Python3_Interpreter_FOUND:     " ${Python3_Interpreter_FOUND})
+message(STATUS "Python3_EXECUTABLE:      " ${Python3_EXECUTABLE})
+message(STATUS "Python3_VERSION:         " ${Python3_VERSION_MAJOR}.${Python3_VERSION_MINOR}.${Python3_VERSION_PATCH})
 
 if (NOT PYTHON_EXTENSION_SUFFIX)
   get_python_extension_suffix()
@@ -150,7 +150,7 @@
 # Detect if the python libs were compiled in debug mode
 # On Linux distros there is no standard way to check that.
 execute_process(
-    COMMAND ${PYTHON_EXECUTABLE} -c "if True:
+    COMMAND ${Python3_EXECUTABLE} -c "if True:
         is_py_debug = False
         import sys
         try:
@@ -176,7 +176,7 @@
 # Detect if python interpeter was compiled with COUNT_ALLOCS define
 # Linux distros are inconsistent in setting the sysconfig.get_config_var('COUNT_ALLOCS') value
 execute_process(
-    COMMAND ${PYTHON_EXECUTABLE} -c "if True:
+    COMMAND ${Python3_EXECUTABLE} -c "if True:
         count_allocs = False
         import sys
         try:
@@ -240,10 +240,10 @@
                   -P "${CMAKE_CURRENT_BINARY_DIR}/cmake_uninstall.cmake")
 
 add_subdirectory(libshiboken)
-add_subdirectory(doc)
+#add_subdirectory(doc)
 
 # deps found, compile the generator.
-if (Qt${QT_MAJOR_VERSION}Core_FOUND AND PYTHONINTERP_FOUND)
+if (Qt${QT_MAJOR_VERSION}_FOUND AND Python3_Interpreter_FOUND)
     add_subdirectory(generator)
     add_subdirectory(shibokenmodule)
 
diff -aru sources/shiboken2/data/shiboken_helpers.cmake sources/shiboken2/data/shiboken_helpers.cmake
--- sources/shiboken2/data/shiboken_helpers.cmake	2024-07-24 13:29:27.437064700 -0700
+++ sources/shiboken2/data/shiboken_helpers.cmake	2024-07-29 12:24:51.763010400 -0700
@@ -5,7 +5,7 @@
 
   # Append a "v" to disambiguate the python version and the shiboken version in the
   # shared library file name.
-  if (APPLE AND PYTHON_VERSION_MAJOR EQUAL 2)
+  if (APPLE AND Python3_VERSION_MAJOR EQUAL 2)
       set(PYTHON_SHARED_LIBRARY_SUFFIX "${PYTHON_SHARED_LIBRARY_SUFFIX}v")
   endif()
 endmacro()
@@ -87,10 +87,11 @@
 
 macro(set_python_site_packages)
     execute_process(
-        COMMAND ${PYTHON_EXECUTABLE} -c "if True:
-            from distutils import sysconfig
+        COMMAND ${Python3_EXECUTABLE} -c "if True:
+            import sysconfig
             from os.path import sep
-            print(sysconfig.get_python_lib(1, 0, prefix='${CMAKE_INSTALL_PREFIX}').replace(sep, '/'))
+            tmp = sysconfig.get_path('platlib').replace(sysconfig.get_path('platstdlib'), '')
+            print(('${CMAKE_INSTALL_PREFIX}'+ tmp).replace(sep, '/') )
             "
         OUTPUT_VARIABLE PYTHON_SITE_PACKAGES
         OUTPUT_STRIP_TRAILING_WHITESPACE)
@@ -103,12 +104,12 @@
 endmacro()
 
 macro(set_python_config_suffix)
-  if (PYTHON_VERSION_MAJOR EQUAL 2)
-      set(PYTHON_CONFIG_SUFFIX "-python${PYTHON_VERSION_MAJOR}.${PYTHON_VERSION_MINOR}")
+  if (Python3_VERSION_MAJOR EQUAL 2)
+      set(PYTHON_CONFIG_SUFFIX "-python${Python3_VERSION_MAJOR}.${Python3_VERSION_MINOR}")
       if (PYTHON_EXTENSION_SUFFIX)
           set(PYTHON_CONFIG_SUFFIX "${PYTHON_CONFIG_SUFFIX}${PYTHON_EXTENSION_SUFFIX}")
       endif()
-  elseif (PYTHON_VERSION_MAJOR EQUAL 3)
+  elseif (Python3_VERSION_MAJOR EQUAL 3)
       if (PYTHON_LIMITED_API)
           if(WIN32)
               set(PYTHON_EXTENSION_SUFFIX "")
@@ -117,7 +118,7 @@
           endif()
           set(PYTHON_CONFIG_SUFFIX ".abi3")
       else()
-          set(PYTHON_CONFIG_SUFFIX "${PYTHON_EXTENSION_SUFFIX}")
+          set(PYTHON_CONFIG_SUFFIX "${Python3_SOABI}")
       endif()
   endif()
 endmacro()
@@ -227,7 +228,7 @@
   # We pick the first most detailed one.
 
   execute_process(
-    COMMAND ${PYTHON_EXECUTABLE} -c "if True:
+    COMMAND ${Python3_EXECUTABLE} -c "if True:
        import re
        try:
            from importlib import machinery
@@ -249,7 +250,7 @@
 
 macro(get_llvm_config)
   execute_process(
-    COMMAND ${PYTHON_EXECUTABLE} -c "if True:
+    COMMAND ${Python3_EXECUTABLE} -c "if True:
        import os
        import sys
        sys.path.append(os.path.realpath(os.path.join('${CMAKE_CURRENT_LIST_DIR}', '..', '..')))
@@ -265,7 +266,7 @@
 
 macro(get_python_arch)
   execute_process(
-    COMMAND ${PYTHON_EXECUTABLE} -c "if True:
+    COMMAND ${Python3_EXECUTABLE} -c "if True:
        import sys
        print('64' if sys.maxsize > 2**31-1 else '32')
        "
@@ -282,13 +283,13 @@
 endmacro()
 
 macro(shiboken_check_if_limited_api)
-    # On Windows, PYTHON_LIBRARIES can be a list. Example:
+    # On Windows, Python3_LIBRARIES can be a list. Example:
     #    optimized;C:/Python36/libs/python36.lib;debug;C:/Python36/libs/python36_d.lib
     # On other platforms, this result is not used at all.
     execute_process(
-        COMMAND ${PYTHON_EXECUTABLE} -c "if True:
+        COMMAND ${Python3_EXECUTABLE} -c "if True:
             import os
-            for lib in '${PYTHON_LIBRARIES}'.split(';'):
+            for lib in '${Python3_LIBRARIES}'.split(';'):
                 if '/' in lib and os.path.isfile(lib):
                     prefix, py = lib.rsplit('/', 1)
                     if py.startswith('python3'):
@@ -299,12 +300,12 @@
         OUTPUT_STRIP_TRAILING_WHITESPACE)
 
     if(FORCE_LIMITED_API STREQUAL "yes")
-        if (${PYTHON_VERSION_MAJOR} EQUAL 3 AND ${PYTHON_VERSION_MINOR} GREATER 4)
+        if (${Python3_VERSION_MAJOR} EQUAL 3 AND ${Python3_VERSION_MINOR} GREATER 4)
             # GREATER_EQUAL is available only from cmake 3.7 on. We mean python 3.5 .
             set(PYTHON_LIMITED_API 1)
         endif()
         if(WIN32)
-            if (${PYTHON_VERSION_MAJOR} EQUAL 3 AND ${PYTHON_VERSION_MINOR} GREATER 4)
+            if (${Python3_VERSION_MAJOR} EQUAL 3 AND ${Python3_VERSION_MINOR} GREATER 4)
                 # PYSIDE-560: XXX maybe add an option to setup.py as override
                 set(SHIBOKEN_PYTHON_LIBRARIES ${PYTHON_LIMITED_LIBRARIES})
             endif()
@@ -321,18 +322,19 @@
         # If no version is specified, just use any interpreter that can be found (from PATH).
         # This is useful for super-project builds, so that the default system interpeter
         # gets picked up (e.g. /usr/bin/python and not /usr/bin/python2.7).
-        find_package(PythonInterp REQUIRED)
-        find_package(PythonLibs REQUIRED)
+        #find_package(PythonInterp REQUIRED)
+        #find_package(PythonLibs REQUIRED)
+        find_package(Python3 COMPONENTS Interpreter Development REQUIRED)
     endif()
     shiboken_validate_python_version()
 
-    set(SHIBOKEN_PYTHON_INTERPRETER "${PYTHON_EXECUTABLE}")
-    set_property(GLOBAL PROPERTY SHIBOKEN_PYTHON_INTERPRETER "${PYTHON_EXECUTABLE}")
+    set(SHIBOKEN_PYTHON_INTERPRETER "${Python3_EXECUTABLE}")
+    set_property(GLOBAL PROPERTY SHIBOKEN_PYTHON_INTERPRETER "${Python3_EXECUTABLE}")
 endmacro()
 
 macro(shiboken_validate_python_version)
-    if((PYTHON_VERSION_MAJOR EQUAL "2" AND PYTHON_VERSION_MINOR LESS "7") OR
-       (PYTHON_VERSION_MAJOR EQUAL "3" AND PYTHON_VERSION_MINOR LESS "5"))
+    if((Python3_VERSION_MAJOR EQUAL "2" AND Python3_VERSION_MINOR LESS "7") OR
+       (Python3_VERSION_MAJOR EQUAL "3" AND Python3_VERSION_MINOR LESS "5"))
             message(FATAL_ERROR
                    "Shiboken requires Python 2.7+ or Python 3.5+.")
     endif()
@@ -353,14 +355,14 @@
     if (SHIBOKEN_COMPUTE_INCLUDES_IS_CALLED_FROM_EXPORT)
         #TODO target_include_directories works on imported targets only starting with v3.11.0.
         set_property(TARGET Shiboken2::libshiboken
-                     APPEND PROPERTY INTERFACE_INCLUDE_DIRECTORIES ${PYTHON_INCLUDE_DIRS})
+                     APPEND PROPERTY INTERFACE_INCLUDE_DIRECTORIES ${Python3_INCLUDE_DIRS})
     else()
         target_include_directories(libshiboken
-                                   PUBLIC $<BUILD_INTERFACE:${PYTHON_INCLUDE_DIRS}>)
+                                   PUBLIC $<BUILD_INTERFACE:${Python3_INCLUDE_DIRS}>)
     endif()
 
 
-    set(SHIBOKEN_PYTHON_INCLUDE_DIRS "${PYTHON_INCLUDE_DIRS}")
+    set(SHIBOKEN_PYTHON_INCLUDE_DIRS "${Python3_INCLUDE_DIRS}")
     message(STATUS
             "SHIBOKEN_PYTHON_INCLUDE_DIRS computed to value: '${SHIBOKEN_PYTHON_INCLUDE_DIRS}'")
 endmacro()
@@ -410,7 +412,7 @@
 
 macro(shiboken_compute_python_libraries)
     shiboken_parse_all_arguments(
-        "SHIBOKEN_COMPUTE_LIBS" "shiboken_compute_python_libraries"
+        "SHIBOKEN_COMPUTE_LIBS" "shiboken_compute_Python3_LIBRARIES"
         "IS_CALLED_FROM_EXPORT" "" "" ${ARGN})
 
     if (NOT SHIBOKEN_PYTHON_LIBRARIES)
@@ -419,13 +421,13 @@
 
     if(CMAKE_BUILD_TYPE STREQUAL "Debug")
         if(WIN32 AND NOT SHIBOKEN_PYTHON_LIBRARIES)
-            set(SHIBOKEN_PYTHON_LIBRARIES ${PYTHON_DEBUG_LIBRARIES})
+            set(SHIBOKEN_PYTHON_LIBRARIES ${Python3_LIBRARIES})
         endif()
     endif()
 
     if(CMAKE_BUILD_TYPE STREQUAL "Release")
         if(WIN32 AND NOT SHIBOKEN_PYTHON_LIBRARIES)
-            set(SHIBOKEN_PYTHON_LIBRARIES ${PYTHON_LIBRARIES})
+            set(SHIBOKEN_PYTHON_LIBRARIES ${Python3_LIBRARIES})
         endif()
     endif()
 
@@ -457,20 +459,20 @@
 endmacro()
 
 function(shiboken_check_if_built_and_target_python_are_compatible)
-    if(NOT SHIBOKEN_PYTHON_VERSION_MAJOR STREQUAL PYTHON_VERSION_MAJOR)
+    if(NOT SHIBOKEN_Python3_VERSION_MAJOR STREQUAL Python3_VERSION_MAJOR)
         message(FATAL_ERROR "The detected Python major version is not \
 compatible with the Python major version which was used when Shiboken was built.
-Built with: '${SHIBOKEN_PYTHON_VERSION_MAJOR}.${SHIBOKEN_PYTHON_VERSION_MINOR}' \
-Detected: '${PYTHON_VERSION_MAJOR}.${PYTHON_VERSION_MINOR}'")
+Built with: '${SHIBOKEN_Python3_VERSION_MAJOR}.${SHIBOKEN_Python3_VERSION_MINOR}' \
+Detected: '${Python3_VERSION_MAJOR}.${Python3_VERSION_MINOR}'")
     else()
         if(NOT SHIBOKEN_PYTHON_LIMITED_API
-           AND NOT SHIBOKEN_PYTHON_VERSION_MINOR STREQUAL PYTHON_VERSION_MINOR)
+           AND NOT SHIBOKEN_Python3_VERSION_MINOR STREQUAL Python3_VERSION_MINOR)
             message(FATAL_ERROR
                     "The detected Python minor version is not compatible with the Python minor \
 version which was used when Shiboken was built. Consider building shiboken with \
 FORCE_LIMITED_API set to '1', so that only the Python major version matters.
-Built with: '${SHIBOKEN_PYTHON_VERSION_MAJOR}.${SHIBOKEN_PYTHON_VERSION_MINOR}' \
-Detected: '${PYTHON_VERSION_MAJOR}.${PYTHON_VERSION_MINOR}'")
+Built with: '${SHIBOKEN_Python3_VERSION_MAJOR}.${SHIBOKEN_Python3_VERSION_MINOR}' \
+Detected: '${Python3_VERSION_MAJOR}.${Python3_VERSION_MINOR}'")
         endif()
     endif()
 endfunction()
diff -aru sources/shiboken2/libshiboken/CMakeLists.txt sources/shiboken2/libshiboken/CMakeLists.txt
--- sources/shiboken2/libshiboken/CMakeLists.txt	2024-07-24 13:29:27.548000200 -0700
+++ sources/shiboken2/libshiboken/CMakeLists.txt	2024-07-29 12:24:51.763010400 -0700
@@ -2,7 +2,7 @@
 
 macro(get_numpy_location)
     execute_process(
-      COMMAND ${PYTHON_EXECUTABLE} -c "if True:
+      COMMAND ${Python3_EXECUTABLE} -c "if True:
       import sys
       import os
       numpy = ''
@@ -49,7 +49,7 @@
 add_custom_command(
     OUTPUT  "${CMAKE_CURRENT_BINARY_DIR}/embed/signature_bootstrap_inc.h"
     OUTPUT  "${CMAKE_CURRENT_BINARY_DIR}/embed/signature_inc.h"
-    COMMAND ${PYTHON_EXECUTABLE} -E
+    COMMAND ${Python3_EXECUTABLE} -E
             "${CMAKE_CURRENT_SOURCE_DIR}/embed/embedding_generator.py"
             --cmake-dir "${CMAKE_CURRENT_BINARY_DIR}/embed"
             --limited-api ${PYTHON_LIMITED_API}
